activerecord/test/cases/batches_test.rb

```diff
@@ -6,7 +6,8 @@ class EachTest < ActiveRecord::TestCase
   fixtures :posts, :subscribers
 
   def setup
-    @posts = Post.order("id asc")
+    Post.primary_key = 'id'
```

Comment:
Instead of setting the primary_key here we should se the primary_key to the old value in each test that changes it in a `ensure` block